src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,38 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.concurrent.locks.ReadWriteLock;
 
-public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+public class ReadWriteList<T> {
+    private final List<T> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-    public void add(E element) {
-        // write your code here
+    public void add(T item) {
+        lock.writeLock().lock();
+        try {
+            list.add(item);
+            System.out.println("Added: " + item);
```

Comment:
Using `System.out.println` for logging is not recommended in production code as it is not thread-safe and cannot be easily managed or formatted. Consider using a proper logging framework like SLF4J or Log4J.